File: golang/206_reverse_linked_list.go
package main

// ListNode Definition for singly-linked list.
type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}
	n, c := head.Next, head
	for n != nil {
		c.Next = n.Next
		n.Next = head
		head = n
		n = c.Next
	}
	return head
}

File: python/206_reverse_linked_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        p = head
        c = head
        if head == None or head.next == None:
            return head
        n = head.next
        while n:
            c.next = n.next
            n.next = p
            p = n
            n = c.next
        return p